File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string compare(string a, string b){
        int n=a.length(), m=b.length();
        string str;
        for(int i=0;i<min(n,m);i++){
            if(a[i]==b[i]){
                str+=a[i];
            } else break;
        }
        return str;
    }
    string longestCommonPrefix(vector<string>& strs) {
        int n=strs.size();
        if(strs.empty()) return "";
        string s=strs[0];
        for(int i=1;i<n;i++){
            s=compare(strs[i],s);
            if(s.empty()) break;
        }
        return s;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n=strs.length;
        if(n==0 || strs==null) return ""; 
        String res=strs[0];
        for(int i=1;i<n;i++){
            res=prefix(strs[i],res);
            if(res.isEmpty()) return "";
        }
        return res;
    }
    private String prefix(String str1, String str2){
        int n=str1.length();
        int m=str2.length();
        int len=Math.min(n,m);
        int i=0;
        while(i<len && str1.charAt(i)==str2.charAt(i)){
            i++;
        }
        return str1.substring(0,i);
    }
}